### DISCUSSION TITLE
function/macro ergonomics

### DISCUSSION THREAD

**Discussion by PierreR** [cue-lang/cue#1096] [Q&A]
It looks like the suggested syntax for function/macro using the `_` has some limit:

For instance it doesn't seem to be possible to do:

```
_stripId: {
    #in: string
    let List = strings.Split(#in, "/")
    List[len(List)-1]
}

// KO
test: _stripId & {_, #in: "hello/world/you"}
```

In that case, it looks like you need to use a `out` field which is less ergonomic:

```
_stripId: {
    #in: string
    let List = strings.Split(#in, "/")
    out: List[len(List)-1]
}

// OK
test: (_stripId1 & {#in: "hello/world/you"}).out
```

https://cuelang.org/play/?id=tYfNk7GSUms#cue@export@cue

**Comment by myitcv** [comment]
Hmm, @PierreR this appears to be a bug. I will raise an issue and link to it from here.

**Reply by myitcv** [reply]
Raised as https://github.com/cue-lang/cue/issues/1116
